app/controllers/v1/open_api_specification_controller.rb

```diff
@@ -0,0 +1,184 @@
+      operation :get do
+        key :description, 'Retrieve vocabulary by string_key'
+        parameter name: :string_key, in: :path, type: :string,
+                  description: 'String identifier for each vocabulary',
+                  pattern: Vocabulary::ALPHANUMERIC_UNDERSCORE_KEY_REGEX,
+                  required: true
+      end
+
+      operation :patch do
+        key :description, 'Update vocabulary'
+
+        parameter name: :string_key, in: :path, type: :string,
+                  required: true
+
+        parameter name: :label, in: :query, type: :string,
+                  required: false
+      end
+
+      operation :delete do
+        key :description, 'Delete vocabulary'
+        parameter name: :string_key, in: :path, type: :string,
+                  required: true
+      end
+    end
+
+    swagger_path '/vocabularies/{string_key}/custom_fields' do
+      operation :post do
+        key :description, 'Create a custom field for a vocabulary'
+        parameter name: :string_key, in: :path, type: :string,
+                  required: true
+
+        parameter name: :field_key, in: :query, type: :string,
+                  required: true, pattern: Vocabulary::ALPHANUMERIC_UNDERSCORE_KEY_REGEX
+
+        parameter name: :label, in: :query, type: :string,
+                  required: true
+
+        parameter name: :data_type, in: :query, type: :string,
+                  required: true, enum: ['string', 'number', 'boolean']
+      end
+    end
+
+    swagger_path '/vocabularies/{string_key}/custom_fields/{field_key}' do
+      operation :patch do
+        parameter name: :string_key, in: :path, type: :string,
+                  required: true
+
+        parameter name: :field_key, in: :path, type: :string,
+                  required: true
+
+        parameter name: :label, in: :query, type: :string
+      end
+
+      operation :delete do
+        parameter name: :string_key, in: :path, type: :string,
+                  required: true
+
+        parameter name: :field_key, in: :path, type: :string,
+                  required: true
+      end
+    end
+
+    Vocabulary.all.each do |v|
+      swagger_path "/vocabularies/#{v.string_key}/terms" do # have to generated for each vocabulary
+        operation :get do # search
+          key :description, "Search for terms within a #{v.string_key} vocabulary"
+
+          parameter name: :per_page, in: :query, type: :number,
+                    default: 20, minimum: 1, maximum: 500,
+                    description: 'Number of results per page'
+
+          parameter name: :page, in: :query, type: :number,
+                    default: 1, minimum: 1,
+                    description: 'Page of paginated results'
+
+          parameter name: :q, in: :query, type: :string
+          parameter name: :uri, in: :query, type: :string
+          parameter name: :pref_label, in: :query, type: :string
+          parameter name: :alt_label, in: :query, type: :string
```

Comment:
Should alt_label be an array instead of a string?

Reply:
Yes!